substrate/client/consensus/beefy/Cargo.toml

```diff
@@ -41,7 +41,7 @@ sp-keystore = { path = "../../../primitives/keystore" }
 sp-mmr-primitives = { path = "../../../primitives/merkle-mountain-range" }
 sp-runtime = { path = "../../../primitives/runtime" }
 tokio = "1.22.0"
-
+smart-default = "0.6.0"
```

Comment:
is this used anywhere? can we avoid it? I would like to reduce external dependencies where possible

```suggestion
```

Reply:
It's used where a struct generic on AuthorityId, needs to implement `Default` like [here](https://github.com/w3f/polkadot-sdk//blob/skalman--beefy-client-generic-on-auth-id/substrate/client/consensus/beefy/src/round.rs#L37).

`PublicKey` doesn't implement `Default` so it couldn't be expected from AuthorityId to be bounded by `Default`. Nonetheless, it shouldn't matter for a struct generic on `AuthorityId` to implement `Default` as long as it does not have a member of `AuthorityId` type (like the one linked above). However, Rust native `Default` macro is not smart enough to understand that and refuses to derive `Default` for such struct. `SmartDefault` on the other hand is able to handle such situation.

The alternative is to implement Default for all such structs manually (and instantiate each of their elements in default funciton), which adds too much verbosity and confusion and reduces elegance just to save on one depedancy, in my opinion.